src/components/ha-selector/ha-selector-number.ts

```diff
@@ -52,12 +52,17 @@ export class HaNumberSelector extends LitElement {
                 : ""}
               <ha-slider
                 labeled
-                .min=${this.selector.number?.min}
-                .max=${this.selector.number?.max}
+                .min=${this.selector.number!.min}
+                .max=${this.selector.number!.max}
                 .value=${this.value ?? ""}
-                .step=${this.selector.number?.step === "any"
-                  ? undefined
-                  : this.selector.number?.step ?? 1}
+                .step=${this.selector.number!.step === "any"
+                  ? Math.min(
+                      1,
+                      (this.selector.number!.max! -
+                        this.selector.number!.min!) /
+                        100
```

Comment:
Is 100 arbitrary? What was the behavior for the old slider?

Reply:
It was a choice from me, to make it 100 steps if it is not specified. I think the old one did something similar, let me check.

Reply:
I think the old one would set the step to 1, but that makes no sense if the max is below 1.

Reply:
IMO, it should be 1, 0.1, 0.01 based on the min/max. 
Because, by only dividing by 100,  if you have min = 0 and max = 1.2, you will have 0.012 and it doesn't make sense.

Reply:
OK, updated